com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/resources/ShutdownResource.java

```diff
@@ -0,0 +1,92 @@
+package com.sap.cloud.lm.sl.cf.web.resources;
+
+import java.text.MessageFormat;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.core.flowable.FlowableFacade;
+import com.sap.cloud.lm.sl.cf.core.message.Messages;
+import com.sap.cloud.lm.sl.cf.core.shutdown.model.ShutdownDto;
+
+@Component
+@Produces(MediaType.APPLICATION_JSON)
+public class ShutdownResource {
+
+    @Inject
+    private FlowableFacade flowableFacade;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ShutdownResource.class);
+
+    static final String ACTION = "shutdown";
+
+    private static final long DEFAULT_COOLDOWN_TIMEOUT_IN_SECONDS = 300L; // 5 mins
+    private long cooldownTimeoutInSeconds = DEFAULT_COOLDOWN_TIMEOUT_IN_SECONDS;
+
+    @POST
+    public ShutdownDto shutdownFlowableProcessEngine(@HeaderParam("x-cf-applicationid") String appId,
+        @HeaderParam("x-cf-instanceid") String appInstanceId, @HeaderParam("x-cf-instanceindex") String appInstanceIndex,
+        @QueryParam("cooldownTimeoutInSeconds") String cooldownTimeoutInSecondsParam) {
+
+        boolean isFlowableProcessEngineActive = flowableFacade.isProcessEngineActive();
+
+        if (!isFlowableProcessEngineActive) {
+            return new ShutdownDto.Builder().isActive(isFlowableProcessEngineActive)
+                .appId(appId)
+                .appInstanceId(appInstanceId)
+                .appInstanceIndex(appInstanceIndex)
+                .cooldownTimeoutInSeconds(cooldownTimeoutInSeconds)
+                .build();
+        }
+
+        if (cooldownTimeoutInSecondsParam != null) {
```

Comment:
extract this if to a method:
private Long getCoolDownTimeout(Long cooldownTimeoutParameter){
   if (cooldownTimeoutParameter == null) return DEFAULT_COOLDOWN_TIMEOUT_IN_SECONDS;
   else return Long.parseLong(cooldownTimeoutInSecondsParam);
}